### ISSUE TITLE
Rich Text Editor line breaks <br> and indentation

### ISSUE THREAD

**Issue by stefkol** [vaadin/flow-components#5183]
### Describe your motivation

We're migrating from Vaadin 8 to Vaadin 24.
In our application RichTextArea (Vaadin 8) was used for writing contracts!
It is important to have the possibility to

- add `<br>` line breaks (new line without a new paragraph)
- create indentation in text or in enumerations (`<ol>, <ul>`)

We're using the html-Values from RichText to print in PDF with respective paragraph or line break.

It is important that the printout in new version is the same as in old version.

The new Rich Text Editor (Vaadin 24) does not provide these options. For example <br> is replaced by new paragraph <p>

### Describe the solution you'd like

In Rich Text Editor (V24) we want to have the possibility to

- add `<br>` line breaks (new line without a new paragraph)
- create indentation in text or in enumerations (`<ol>, <ul>`)

**Without these opportunities we are not allowed to migrate our application!**

**Comment by ghe-bm** [comment]
Hello,
we are also migrating from Vaadin 8 to 23/24 and are using the component in a similar way.
In additon to the <br>, we also need the following formatting features to migrate to Vaadin 23 (in priority order):

- font styles (background, foreground, font selection, size) -> font-tag
- horizontal rule -> hr-tag
- indentation -> blockquote with margin

I am a suprised that a prime component does not have at minimum the features of the Vaadin 8 rich text editor. We need these features because the current version of our application persists the generated HTML in the database and if we open the old HTML it would not work in the new rich text editor version. And we do not want to loose formatting information.

And we need a migration guide on how we have to convert the old HTML structure to the new HTML structure if it is note the same.

By the way, I did not find a migration guide for the rich text editor in your documentation.
I attached a file with a generated HTML code example from Vaadin 8 rich text editor.
[generated_html_example.txt](https://github.com/vaadin/flow-components/files/11904130/generated_html_example.txt)

**Comment by ghe-bm** [comment]
Hello,
I ask myself why this issue is classified as an enhancement?
From my point of view these are features that exists before (in Vaadin 8) and now are missing.
So it should be classified as a bug.
Additionally the rich text editor component is the main reason, why our company bought Pro licenses for Vaadin.

Could you please clarify this.

**Comment by web-padawan** [comment]
> From my point of view these are features that existed before (in Vaadin 8) and now are missing.
> So it should be classified as a bug.

Thanks for your feedback. Please note that the underlying implementation is totally different and is based on [QuillJS](https://github.com/quilljs/quill) library. 

Indentation and nested lists are not currently supported because of the fact that Quill lacks support for it, as discussed in https://github.com/vaadin/web-components/issues/1610. So technically it's a missing feature, and implementing it is non-trivial.

**Comment by ghe-bm** [comment]
And what about font type, size, background, foreground? I have seen that these features are supported by Quill.

**Comment by web-padawan** [comment]
Yes, some of [Quill formats](https://quilljs.com/docs/formats/) are currently not supported in our [Quill fork](https://github.com/vaadin/quill). Not all of them are straightforward to implement due to having to work with Quill internals. We could consider adding font features but it would need some research.

**Comment by ghe-bm** [comment]
@stefkol Hello Stefkol, did you find another solution to your Problem? Are you using a vaadin editor add-on like TnyMCE or CKeditor? It seems that the Vaadin team is not going to expand the features of the Rich Text Edior anytime soon.

**Comment by stefkol** [comment]
@ghe-bm Yes we're just checking VaadinCkEditor for our needs.
At the moment there are some issues (Error with validation: checkValidity not found AND we don't want to have a valueChanged-Event on every keystroke)

**Comment by web-padawan** [comment]
Web components PRs for indentation / nested lists targeting Vaadin 25 are merged:

- https://github.com/vaadin/web-components/pull/9399
- https://github.com/vaadin/web-components/pull/9971

Just to clarify: the other point in of this issue about adding `<br>` needs to be implemented separately.
If that's about soft line breaks, we could adapt solution from https://github.com/quill-next/quill-next/pull/32.

**Comment by web-padawan** [comment]
Closing in favor of https://github.com/vaadin/flow-components/issues/6302, let's continue discussing `<br>` tags there.
